commit eb2b5489b3f8f72319215b4c2c453bed62f7b9e7
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Jun 14 09:41:07 2011 +0400

    Fix get Uid and Gid by passwd and group files.

diff --git a/pym/cl_template.py b/pym/cl_template.py
index 753d801..8fe5d74 100644
--- a/pym/cl_template.py
+++ b/pym/cl_template.py
@@ -4329,9 +4329,11 @@ variable 'cl_name'"))
         """Get uid by username from chroot passwd file"""
         passwdFile = os.path.join(self._baseDir,'etc/passwd')
         if os.path.exists(passwdFile):
-            mapUid = dict(map(lambda x:x.split(':')[0:3:2],
+            mapUid = dict(
+                     filter(lambda x:x[0] and x[1],
+                     map(lambda x:x.split(':')[0:3:2],
                      filter(lambda x:not x.startswith('#'),
-                     open(passwdFile,'r'))))
+                     open(passwdFile,'r')))))
             if strUid in mapUid:
                 return int(mapUid[strUid])
         return None
@@ -4340,9 +4342,11 @@ variable 'cl_name'"))
         """Get gid by groupname from chroot group file"""
         groupFile = os.path.join(self._baseDir,'etc/group')
         if os.path.exists(groupFile):
-            mapGid = dict(map(lambda x:x.split(':')[0:3:2],
+            mapGid = dict(
+                     filter(lambda x:x[0] and x[1],
+                     map(lambda x:x.split(':')[0:3:2],
                      filter(lambda x:not x.startswith('#'),
-                     open(groupFile,'r'))))
+                     open(groupFile,'r')))))
             if strGid in mapGid:
                 return int(mapGid[strGid])
         return None
